app/src/main/java/com/github/se/project/ui/components/LessonEditor.kt

```diff
@@ -620,10 +651,8 @@ fun validateLessonInput(
  * @return `true` if the lesson has an instant status, `false` otherwise.
  */
 fun isInstant(lesson: Lesson?): Boolean {
-  return if (lesson == null) false
-  else
-      (lesson.status == LessonStatus.INSTANT_REQUESTED) ||
-          (lesson.status == LessonStatus.INSTANT_CONFIRMED)
+  return (lesson?.status == LessonStatus.INSTANT_REQUESTED) || // Instant request status
```

Comment:
The use of '?' makes the function more concise, but personally I find the older version slightly more readable, as it makes it clearer how the null handling works